[WIP] get_entity_type: enforce disagreements with Wikidata P31 typing

maxlath opened the pull request:
especially,
- an entity can't be an edition if it doesn't have `wdt:P629` claims
- prefer to consider an ambiguous entity as a serie rather than a work
  as it will be then easier to create subparts of this entity to which to link editions
  In the worst case, that was actually not a serie and subparts can be merged into it

maxlath committed:
get_entity_type: enforce disagreements with Wikidata P31 typing

especially,
- an entity can't be an edition if it doesn't have wdt:P629 claims
- prefer to consider an ambiguous entity as a serie rather than a work
  as it will be then easier to create subparts of this entity to which to link editions
  In the worst case, that was actually not a serie and subparts can be merged into it

jum-s commented on server/controllers/entities/lib/get_entity_type.coffee:
```diff
@@ -2,23 +2,49 @@ __ = require('config').universalPath
+      isSerie = 'serie' in propertyTypes
+
+      # If it has 'edition of' claims, consider that it's an edition
+      if isEdition and claims['wdt:P629']? then return 'edition'
```
We might consider returning a `undefined` value as no wikidata edition are to be interpreted by inventaire has an edition. Inventaire would then consider this entity as a topic, not an edition, avoiding for example the possibility of adding it to a user's inventaire.

jum-s commented:
Some code to [refactor](https://github.com/inventaire/inventaire/blob/cd065234d702560ac865e0a2a6c5363c25314517/server/controllers/entities/lib/format_edition_entity.coffee#L8)

maxlath committed:
radically enforce wikidata editions quarantine

by returning undefined instead of edition for wd entities

- items that reference wikidata edition would need to be updated to reference
  their inv equivalents
- that means keeping duplicated data with Wikidata, which is a bit sad.
  Implementing data synchronization would look a hell lot like "shadow items"

The good side of the coin:
- this does make some things simpler: looking for the editions of a work
  can be done on Inventaire only (which is already the case at this point
  but it wasn't clear if we would add Wikidata lookup at some point)

maxlath commented:
@jum-s how would you like to see [it](https://github.com/inventaire/inventaire/blob/cd065234d702560ac865e0a2a6c5363c25314517/server/controllers/entities/lib/format_edition_entity.coffee#L8) refactored?

maxlath commented:
@jum-s as you know I'm having doubts about the whole thing, please do not merge yet
